server/services/auth_service.py

```
@@ -74,6 +83,14 @@ async def login(self, login_data: LoginRequest) -> LoginResponse:
             raise
         except AuthError:
             raise
+        except HTTPException as e:
+            # Re-raise HTTPException from repository as AuthError to maintain service layer abstraction
+            if e.status_code == 401:
+                raise AuthError(e.detail)
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)